src/game_battler.cpp

```diff
@@ -478,84 +478,46 @@ static int AffectParameter(const int type, const int val) {
 		val;
 }
 
-int Game_Battler::GetAtk(int weapon) const {
-	int base_atk = GetBaseAtk(weapon);
-	int n = Utils::Clamp(base_atk, 1, MaxStatBaseValue());
-
-	for (int16_t i : GetInflictedStates()) {
-		// States are guaranteed to be valid
-		const lcf::rpg::State& state = *lcf::ReaderUtil::GetElement(lcf::Data::states, i);
-		if (state.affect_attack) {
-			n = AffectParameter(state.affect_type, base_atk);
-			break;
+static int AdjustParam(int base, int mod, int maxval, Span<const int16_t> states, bool lcf::rpg::State::*adj) {
```

Comment:
> Refactor GetAtk() etc.. 

What is this ``bool lcf::rpg::State::*adj``? Looks like a Pointer to ``State::``... what?

Reply:
It's a pointer to data member, points to `State::affect_atack` or `State::affect_defense` etc.. Essentially lets me write one function for all 4 parameters without needing templates.